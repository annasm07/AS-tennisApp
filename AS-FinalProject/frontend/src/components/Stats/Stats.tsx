import React, {useEffect} from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {
  Text,
  View,
  Image,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import {
  getStatsInfo,
  getPlayerInfo,
  deleteMatch,
} from '../../redux/actions/actionCreators';
import StatsBar from '../StatsBars/StatsBars';
import matchBoxStyles from '../../theme/matchBoxTheme';

export default function Stats({route, navigation}: any) {
  const {matchId} = route.params;
  const dispatch = useDispatch();

  const tokens = useSelector((store: any) => store.tokens);
  const user = useSelector((store: any) => store.user);
  let {stats} = useSelector((store: any) => store.selectedStats);
  let {match} = useSelector((store: any) => store.selectedStats);

  useEffect(() => {
    dispatch(getStatsInfo(tokens[0], matchId));
  }, [dispatch, tokens, matchId]);
  function handleDelete() {
    dispatch(deleteMatch(tokens[0], matchId));
    dispatch(getPlayerInfo(tokens[0], user.user.playerId));
    navigation.navigate('Dashboard');
  }
  return (
    <SafeAreaView>
      <Text style={styles.title}>Match Details</Text>
      <ScrollView>
        {match && (
          <>
            <View style={matchBoxStyles.matchBox}>
              <View style={matchBoxStyles.player1}>
                <Text>{match?.result[0].name}</Text>
                <Text style={matchBoxStyles.result}>
                  {match?.result[0].games}
                </Text>
              </View>
              <View style={matchBoxStyles.player2}>
                <Text>{match?.result[1].name}</Text>
                <Text style={matchBoxStyles.result}>
                  {match?.result[1].games}
                </Text>
              </View>
            </View>
            <View style={styles.names}>
              <Text style={styles.names}>{match?.result[0].name}</Text>
              <Text style={styles.names}>{match?.result[1].name}</Text>
            </View>
          </>
        )}

        {stats?.length ? (
          <>
            <StatsBar
              title={'Aces'}
              p1Value={stats[0]?.serve.aces}
              p2Value={stats[1]?.serve.aces}
            />
            <StatsBar
              title={'Double Faults'}
              p1Value={stats[0]?.serve.doubleFaults}
              p2Value={stats[1]?.serve.doubleFaults}
            />
            <StatsBar
              title={'Forced Errors'}
              p1Value={stats[0]?.points.forcedErrors.baseLine[0]}
              p2Value={stats[1]?.points.forcedErrors.baseLine[0]}
            />
            <StatsBar
              title={'Unforced Errors'}
              p1Value={stats[0]?.points.unforcedErrors.baseLine[0]}
              p2Value={stats[1]?.points.unforcedErrors.baseLine[0]}
            />
            <StatsBar
              title={'Winners'}
              p1Value={stats[0]?.points.winners.baseLine[0]}
              p2Value={stats[1]?.points.winners.baseLine[0]}
            />
            <TouchableOpacity
              style={styles.deleteButton}
              onPress={() => handleDelete()}>
              <Text>Delete</Text>
            </TouchableOpacity>
          </>
        ) : (
          <Image
            source={{
              uri: 'https://www.gatoslechuzos.com/static/loading.gif',
            }}
            style={styles.loading}
          />
        )}
        {match && (
          <>
            <View>
              <Text>{match?.flow.points[0][3].p1}</Text>
            </View>
          </>
        )}
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  loading: {
    width: 200,
    height: 200,
    top: '45%',
    left: '40%',
  },
  title: {
    fontSize: 26,
    paddingLeft: 20,
    marginTop: -25,
  },
  names: {
    borderBottomColor: '#C4C4C4',
    borderBottomWidth: 1,
    fontSize: 24,
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginHorizontal: 10,
    marginTop: 10,
    marginBottom: 5,
  },
  deleteButton: {
    alignItems: 'center',
    alignSelf: 'center',
    backgroundColor: '#F6655B',
    borderRadius: 10,
    fontSize: 50,
    marginTop: 30,
    paddingHorizontal: 12,
    paddingVertical: 10,
    width: 120,
  },
});
