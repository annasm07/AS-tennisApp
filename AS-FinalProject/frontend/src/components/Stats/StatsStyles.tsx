import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  stats: {
    marginVertical: 10,
  },
  statInfo: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '90%',
    alignSelf: 'center',
  },
  statsFullBar: {
    width: '90%',
    alignSelf: 'center',
    backgroundColor: '#E5E5E5',
    height: 14,
    display: 'flex',
    flexDirection: 'row',
    borderRadius: 5,
  },
  statsHalfBar: {
    width: '50%',
  },
  statsPlayer1Bar: {
    width: '50%',
    paddingHorizontal: 20,
    alignSelf: 'flex-end',
    backgroundColor: '#F5BF00',
    height: 14,
    borderRightColor: '#000',
    borderRightWidth: 1,
    borderTopLeftRadius: 5,
    borderBottomLeftRadius: 5,
  },
  statsPlayer2Bar: {
    width: '50%',
    paddingHorizontal: 20,
    alignSelf: 'flex-start',
    backgroundColor: '#F7A65B',
    height: 14,
    borderLeftColor: '#000',
    borderLeftWidth: 1,
    borderTopRightRadius: 5,
    borderBottomRightRadius: 5,
  },
});
export default styles;
