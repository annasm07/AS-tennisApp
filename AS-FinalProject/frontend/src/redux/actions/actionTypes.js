export default {
  LOG_IN: 'LOG_IN',
  LOG_IN_ERROR: 'LOG_IN_ERROR',
  SIGN_UP: 'LOG_IN',
  SIGN_UP_ERROR: 'LOG_IN_ERROR',
  GET_PLAYER: 'GET_PLAYER',
  GET_PLAYER_ERROR: 'GET_PLAYER_ERROR',
  GET_PLAYERS: 'GET_PLAYERS',
  GET_PLAYERS_ERROR: 'GET_PLAYERS_ERROR',
  CREATE_MATCH: 'CREATE_MATCH',
  CREATE_MATCH_ERROR: 'CREATE_MATCH_ERROR',
  UPDATE_POINTS: 'UPDATE_POINTS',
  UPDATE_POINTS_ERROR: 'UPDATE_POINTS_ERROR',
  UPDATE_GAMES: 'UPDATE_GAMES',
  UPDATE_GAMES_ERROR: 'UPDATE_GAMES_ERROR',
  END_GAME: 'END_GAME',
  END_GAME_ERROR: 'END_GAME_ERROR',
  END_SET: 'END_SET',
  END_SET_ERROR: 'END_SET_ERROR',
  END_MATCH: 'END_MATCH',
  END_MATCH_ERROR: 'END_MATCH_ERROR',
};
