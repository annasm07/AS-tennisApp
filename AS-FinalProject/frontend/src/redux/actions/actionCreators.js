import axios from 'axios';
import actionTypes from './actionTypes';

const LOG_IN_URL = 'http://localhost:1616/api/auth/login';
const SIGN_UP_URL = 'http://localhost:1616/api/auth/signup';
const GET_PLAYER_URL = 'http://localhost:1616/api/home';
const GET_ALL_PLAYERS = 'http://localhost:1616/api/players';
const CREATE_MATCH_URL = 'http://localhost:1616/api/match';

export function logIn(email, password) {
  return async dispatch => {
    try {
      const {data} = await axios.post(LOG_IN_URL, {email, password});
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.LOG_IN_ERROR,
      });
    }
  };
}

export function signUp({name, email, password, player, playerName}) {
  return async dispatch => {
    try {
      await axios.post(SIGN_UP_URL, {
        name,
        email,
        password,
        player,
        playerName,
      });
      dispatch({
        // type: actionTypes.SIGN_UP,
        // user: data.user,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.SIGN_UP_ERROR,
      });
    }
  };
}

export function getPlayerInfo(token, playerId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${GET_PLAYER_URL}/${playerId}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYER,
        player: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYER_ERROR,
      });
    }
  };
}
export function getAllPlayers(token) {
  return async dispatch => {
    try {
      const {data} = await axios(GET_ALL_PLAYERS, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYERS,
        players: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYERS_ERROR,
      });
    }
  };
}

export function newMatch(token, p1Name, p2Name) {
  return async dispatch => {
    try {
      const {data} = await axios.post(
        CREATE_MATCH_URL,
        {p1Name, p2Name},
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      );
      dispatch({
        type: actionTypes.CREATE_MATCH,
        currentMatch: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.CREATE_MATCH_ERROR,
      });
    }
  };
}
