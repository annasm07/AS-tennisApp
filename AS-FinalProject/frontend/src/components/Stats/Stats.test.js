import {render} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/actionCreators';
import {Provider} from 'react-redux';
import Stats from './Stats';

jest.mock('../../redux/actions/actionCreators');

describe('When invoked a Dashboard component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    selectedStats: {
      stats: [
        {
          serve: {aces: 4, doubleFaults: 8},
          points: {
            forcedErrors: {baseLine: [4, 9]},
            unforcedErrors: {baseLine: [5, 11]},
            winners: {baseLine: [2, 4]},
          },
        },
        {
          serve: {aces: 4, doubleFaults: 8},
          points: {
            forcedErrors: {baseLine: [5, 11]},
            unforcedErrors: {baseLine: [5, 11]},
            winners: {baseLine: [2, 4]},
          },
        },
      ],
      match: {
        result: [
          {
            name: 'Anna',
            games: [2, 3, 4],
          },
          {
            name: 'Anna',
            games: [2, 3, 4],
          },
        ],
      },
    },
    tokens: ['abc', 'cde'],
    matchId: '123cde',
  });
  let route;
  let matchId;
  let selectedStats = {stats: [], match: {}};
  beforeEach(() => {
    jest.spyOn(actions, 'getStatsInfo').mockReturnValueOnce({type: ''});

    route = {
      params: {matchId: '123cde'},
    };
  });

  test('Should render Stats component', () => {
    const statsPage = render(
      <Provider store={store}>
        <Stats
          route={route}
          matchId={matchId}
          stats={selectedStats.stats}
          match={selectedStats.match}
        />
      </Provider>,
    );
    expect(statsPage).toMatchSnapshot();
  });
});

describe('When invoked a Dashboard component, stats.length = undefined', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    selectedStats: {
      stats: [],
      match: {
        result: [
          {
            name: 'Anna',
            games: [2, 3, 4],
          },
          {
            name: 'Anna',
            games: [2, 3, 4],
          },
        ],
      },
    },
    tokens: ['abc', 'cde'],
    matchId: '123cde',
  });
  let route;
  let matchId;
  let selectedStats = {stats: [], match: {}};
  beforeEach(() => {
    jest.spyOn(actions, 'getStatsInfo').mockReturnValueOnce({type: ''});

    route = {
      params: {matchId: '123cde'},
    };
  });

  test('Should render Stats component', () => {
    const statsPage = render(
      <Provider store={store}>
        <Stats
          route={route}
          matchId={matchId}
          stats={selectedStats.stats}
          match={selectedStats.match}
        />
      </Provider>,
    );
    expect(statsPage).toMatchSnapshot();
  });
});
