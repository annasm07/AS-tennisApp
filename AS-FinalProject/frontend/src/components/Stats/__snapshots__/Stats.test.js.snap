// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a Dashboard component Should render Stats component 1`] = `
<RNCSafeAreaView>
  <Text
    style={
      Object {
        "fontSize": 26,
        "marginTop": -25,
        "paddingLeft": 20,
      }
    }
  >
    Match Details
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#7A7A7A",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "letterSpacing": 20,
          }
        }
      >
        2
        3
        4
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "letterSpacing": 20,
          }
        }
      >
        2
        3
        4
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#C4C4C4",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "fontSize": 24,
        "justifyContent": "space-between",
        "marginBottom": 5,
        "marginHorizontal": 10,
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "borderBottomColor": "#C4C4C4",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "fontSize": 24,
          "justifyContent": "space-between",
          "marginBottom": 5,
          "marginHorizontal": 10,
          "marginTop": 10,
        }
      }
    >
      Anna
    </Text>
    <Text
      style={
        Object {
          "borderBottomColor": "#C4C4C4",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "fontSize": 24,
          "justifyContent": "space-between",
          "marginBottom": 5,
          "marginHorizontal": 10,
          "marginTop": 10,
        }
      }
    >
      Anna
    </Text>
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
          "width": "90%",
        }
      }
    >
      <Text>
        4
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Aces
      </Text>
      <Text>
        4
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "height": 14,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#F5BF00",
                "borderBottomLeftRadius": 5,
                "borderRightColor": "#000",
                "borderRightWidth": 1,
                "borderTopLeftRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F7A65B",
                "borderBottomRightRadius": 5,
                "borderLeftColor": "#000",
                "borderLeftWidth": 1,
                "borderTopRightRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
          "width": "90%",
        }
      }
    >
      <Text>
        8
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Double Faults
      </Text>
      <Text>
        8
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "height": 14,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#F5BF00",
                "borderBottomLeftRadius": 5,
                "borderRightColor": "#000",
                "borderRightWidth": 1,
                "borderTopLeftRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F7A65B",
                "borderBottomRightRadius": 5,
                "borderLeftColor": "#000",
                "borderLeftWidth": 1,
                "borderTopRightRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
          "width": "90%",
        }
      }
    >
      <Text>
        4
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Forced Errors
      </Text>
      <Text>
        5
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "height": 14,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#F5BF00",
                "borderBottomLeftRadius": 5,
                "borderRightColor": "#000",
                "borderRightWidth": 1,
                "borderTopLeftRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "44%",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F7A65B",
                "borderBottomRightRadius": 5,
                "borderLeftColor": "#000",
                "borderLeftWidth": 1,
                "borderTopRightRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "56%",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
          "width": "90%",
        }
      }
    >
      <Text>
        5
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Unforced Errors
      </Text>
      <Text>
        5
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "height": 14,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#F5BF00",
                "borderBottomLeftRadius": 5,
                "borderRightColor": "#000",
                "borderRightWidth": 1,
                "borderTopLeftRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F7A65B",
                "borderBottomRightRadius": 5,
                "borderLeftColor": "#000",
                "borderLeftWidth": 1,
                "borderTopRightRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 10,
          "width": "90%",
        }
      }
    >
      <Text>
        2
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        Winners
      </Text>
      <Text>
        2
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E5E5E5",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "height": 14,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#F5BF00",
                "borderBottomLeftRadius": 5,
                "borderRightColor": "#000",
                "borderRightWidth": 1,
                "borderTopLeftRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#F7A65B",
                "borderBottomRightRadius": 5,
                "borderLeftColor": "#000",
                "borderLeftWidth": 1,
                "borderTopRightRadius": 5,
                "height": 14,
                "paddingHorizontal": 20,
                "width": "50%",
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`When invoked a Dashboard component, stats.length = undefined Should render Stats component 1`] = `
<RNCSafeAreaView>
  <Text
    style={
      Object {
        "fontSize": 26,
        "marginTop": -25,
        "paddingLeft": 20,
      }
    }
  >
    Match Details
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#7A7A7A",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "letterSpacing": 20,
          }
        }
      >
        2
        3
        4
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 8,
        }
      }
    >
      <Text>
        Anna
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
            "letterSpacing": 20,
          }
        }
      >
        2
        3
        4
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#C4C4C4",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "fontSize": 24,
        "justifyContent": "space-between",
        "marginBottom": 5,
        "marginHorizontal": 10,
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "borderBottomColor": "#C4C4C4",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "fontSize": 24,
          "justifyContent": "space-between",
          "marginBottom": 5,
          "marginHorizontal": 10,
          "marginTop": 10,
        }
      }
    >
      Anna
    </Text>
    <Text
      style={
        Object {
          "borderBottomColor": "#C4C4C4",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "fontSize": 24,
          "justifyContent": "space-between",
          "marginBottom": 5,
          "marginHorizontal": 10,
          "marginTop": 10,
        }
      }
    >
      Anna
    </Text>
  </View>
  <Image
    source={
      Object {
        "uri": "https://www.gatoslechuzos.com/static/loading.gif",
      }
    }
    style={
      Object {
        "height": 200,
        "left": "40%",
        "top": "45%",
        "width": 200,
      }
    }
  />
</RNCSafeAreaView>
`;
